public/js/monitor.js

```diff
@@ -136,20 +136,48 @@ function toggleArticles() {
   });
 }
 
-function toggleMobileFeatures() {
-  const windowWidth = window.innerWidth;
+function hideShowNavBars(win, navBar) {
+  win.onscroll = function(e) {
+    if (win.pageYOffset < 100) {
+      navBar.classList = ["show-nav-bars"];
+      return;
+    }
+
+    if (this.oldScroll < this.scrollY) {
+      navBar.classList = ["hide-nav-bars"];
+      this.oldScroll = this.scrollY;
+      return;
+    }
+
+    if (this.oldScroll > this.scrollY + 50) {
+      navBar.classList = ["show-nav-bars"];
+      this.oldScroll = this.scrollY;
+      return;
+    }
+    this.oldScroll = this.scrollY;
+  };
+}
+
+function toggleMobileFeatures(topNavBar) {
+  const win = window;
+  const windowWidth = win.innerWidth;
   if (windowWidth > 800) {
     const emailCards = document.querySelectorAll(".col-9.email-card:not(.zero-breaches)");
       emailCards.forEach(card => {
         card.classList.add("active");
       });
+      win.removeEventListener("scroll", hideShowNavBars);
       return;
     }
 
   const closeActiveEmailCards = document.querySelectorAll(".col-9.email-card.active");
     closeActiveEmailCards.forEach(card => {
       card.classList.remove("active");
     });
+
+    if (windowWidth < 600) {
+      win.addEventListener("scroll", hideShowNavBars(win, topNavBar));
```

Comment:
This adds `hideShowNavBars` as an event listener on window scroll, but `hideShowNavBars` also uses `onscroll` to add an event listener to window scrolling?